desktop-widgets/diveplanner.cpp

```diff
@@ -232,10 +244,42 @@ void DivePlannerWidget::heightChanged(const int height)
 	plannerModel->setSurfacePressure(pressure);
 }
 
-void DivePlannerWidget::salinityChanged(const double salinity)
+void DivePlannerWidget::waterTypeUpdateTexts()
 {
-	/* Salinity is expressed in weight in grams per 10l */
-	plannerModel->setSalinity(lrint(10000 * salinity));
+	double density;
+	for (int i = 0; i < 3; i++) {
+		if (ui.waterType->itemData(i) != QVariant::Invalid) {
+			QString densityText = ui.waterType->itemText(i).split("(")[0].trimmed();
+			density = ui.waterType->itemData(i).toDouble();
+			densityText.append(QString(" (%1kg/l)").arg(QString::number(density, 'f', 2)));
+			ui.waterType->setItemText(i, densityText);
+		}
+	}
+}
+
+void DivePlannerWidget::waterTypeChanged(const int index)
+{
+	bool ok = true;
+	double density;
+	
+	if (index == 2) {
```

Comment:
For extensibility, don't hardcode the index of the last item. Instead, test the data for a special value (e.g. as above `itemData(i) != QVariant::Invalid`).